File: backend/src/AqlanDentalPro.Domain/Entities/Visit.cs
using AqlanDentalPro.Domain.Enums;

namespace AqlanDentalPro.Domain.Entities;

public class Visit : BaseEntity
{
    public Guid PatientId { get; set; }
    public Guid? AppointmentId { get; set; }
    public DateOnly VisitDate { get; set; }
    public string? VisitType { get; set; }
    public Specialty? Specialty { get; set; }
    public Guid? DoctorId { get; set; }
    public string? ChiefComplaint { get; set; }
    public string? ClinicalNotes { get; set; }
    public string? TreatmentDone { get; set; }
    public string? Diagnosis { get; set; }
    public string? Instructions { get; set; }
    public string? NextVisitPlan { get; set; }
    public decimal? Cost { get; set; }
    public DateOnly? NextVisitDate { get; set; }

    // ── Patient Journey fields (Sprint: Command Center) ────────────────────
    /// <summary>Service associated with this visit.</summary>
    public Guid? ServiceId { get; set; }
    /// <summary>Checkout status: null=pending, ReadyForCheckout, CheckedOut.</summary>
    public string? CheckoutStatus { get; set; }
    /// <summary>When the doctor marked the visit as ready for checkout.</summary>
    public DateTime? ReadyForCheckoutAt { get; set; }
    /// <summary>Reference amount due at checkout (not a finance calculation).</summary>
    public decimal? AmountDueReference { get; set; }
    /// <summary>Suggested billing procedure from clinical diagnosis (without price).</summary>
    public string? ProposedProcedure { get; set; }

    public Patient Patient { get; set; } = null!;
    public Appointment? Appointment { get; set; }
    public Doctor? Doctor { get; set; }
    public ICollection<GeneralTreatment> GeneralTreatments { get; set; } = [];
    public ICollection<Prescription> Prescriptions { get; set; } = [];
}

File: .migration-backup/frontend/src/types/visit.ts
/**
 * Shared visit types — extracted from inline definitions.
 * Aligned with backend: GET /api/visits
 */

export interface VisitDto {
  id: string;
  patientId: string;
  appointmentId?: string;
  visitDate: string;
  visitType?: string;
  specialty?: string;
  doctorId?: string;
  doctorName?: string;
  chiefComplaint?: string;
  clinicalNotes?: string;
  treatmentDone?: string;
  diagnosis?: string;
  instructions?: string;
  nextVisitPlan?: string;
  cost?: number;
  nextVisitDate?: string;
  serviceId?: string;
  checkoutStatus?: string;
  readyForCheckoutAt?: string;
  amountDueReference?: number;
  isActive: boolean;
  createdAt: string;
  updatedAt: string;
  appointment?: {
    appointmentDate: string;
    appointmentTime?: string;
    appointmentType?: string;
    appointmentStatus?: string;
    doctorName?: string;
  };
}

export interface CreateVisitRequest {
  patientId: string;
  appointmentId?: string;
  visitDate?: string;
  visitType?: string;
  specialty?: string;
  doctorId?: string;
  serviceId?: string;
  chiefComplaint?: string;
  clinicalNotes?: string;
  treatmentDone?: string;
  diagnosis?: string;
  instructions?: string;
  nextVisitPlan?: string;
  cost?: number;
  nextVisitDate?: string;
}

export type UpdateVisitRequest = Partial<CreateVisitRequest>;

export const VISIT_TYPE_OPTIONS = [
  { value: "Consultation", label: "استشارة" },
  { value: "FollowUp", label: "متابعة" },
  { value: "Emergency", label: "طارئ" },
  { value: "Treatment", label: "علاج" },
  { value: "Review", label: "مراجعة" },
] as const;
